### ISSUE TITLE
Mixing arrow function properties with methods

### ISSUE THREAD

**Issue by the-owl** [bryanrsmith/eslint-plugin-sort-class-members#49]
I haven't found a way to treat methods and arrow function properties as a single group. Is this currently possible? I'd like to treat arrow function properties as methods and sort them alphabetically together, like this:
```
class A {
  a = () => {}

  b () {}

  c = () => {}

  d () {}
}
```

**Comment by bryanrsmith** [comment]
I think you're right. There isn't currently a way to define a group whose matches are unordered. I imagine this could be useful for React components, where methods are often converted to arrow function properties to avoid binding issues.

This is almost it, except it requires methods to come before properties
```js
groups: {
	methodOrFuncProperty: [
		{ type: 'method' },
		{ type: 'property', propertyType: 'ArrowFunctionExpression' },
	],
},
```


There is a `sort` option on a group, so maybe we could make it apply to nested groups?


```js
groups: {
	methodOrFuncProperty: {
		rules: [
			{ type: 'method' },
			{ type: 'property', propertyType: 'ArrowFunctionExpression' },
		],
		sort: 'none'
	},
},
```

**Comment by the-owl** [comment]
> There is a `sort` option on a matcher, but maybe we could move it up to the group level?
> 
> ```js
> groups: {
> 	methodOrFuncProperty: {
> 		rules: [
> 			{ type: 'method' },
> 			{ type: 'property', propertyType: 'ArrowFunctionExpression' },
> 		],
> 		sort: 'none',
> 	},
> },
> ```

Yes, this will solve the problem! I can try to implement this feature and send a PR.

**Comment by nsisodiya** [comment]
Thanks

**Comment by nsisodiya** [comment]
@bryanrsmith - I am confused, above config will work or proposed? 
Please let us know what is the right config if we want to mix (arrow functions and methods) and then sort both together.

**Comment by bryanrsmith** [comment]
No, it's is not currently supported. I was proposing that the `sort` option be supported at the group level to enable this functionality. PRs welcome!

**Comment by nsisodiya** [comment]
@bryanrsmith - may you guide me, I can try for a PR.

**Comment by kazlauskis** [comment]
You can already wrap the arrow functions with `methods`, which allows to keep the methods and the arrow props in the same place. I am not sure why this works but it does for me.

```
"order": [
            "[static-properties]",
            "[static-methods]",
            "[properties]",
            "constructor",
            "[methods]",
            {
              "type": "property",
              "propertyType": "ArrowFunctionExpression"
            },
            "[methods]"
 ],
```

**Comment by piranna** [comment]
Any update on this? I'm also interested. Maybe instead of `rules`, we can reuse the `groups`, so we don't need to define them again, as far as we prevent loops it will be fine.

**Comment by bryanrsmith** [comment]
No updates. I haven't personally used this plugin in a long time, as classes are not a language feature I use much any more. I'm happy to review PRs, but I'm not very motivated to put any effort into new features myself. Sounds like a plan has been proposed earlier in the thread, so feel free to take a stab at it.

**Comment by piranna** [comment]
> classes are not a language feature I use much any more

Just curious, what pattern are you using now? Functional programming? Split of data and procedures?

**Comment by bryanrsmith** [comment]
Yep. I do a lot of React work, and my class use dropped off pretty hard when React introduced hooks in 2019.

**Comment by piranna** [comment]
Ok, make sense. Personally I think hooks were an error, I still prefer classes, but if they are useful to you, that's fine :-)
